### PR TITLE
EASY-2100: add property:  bag-store.bag-name

### PR DESCRIPTION
**Pull request by stephanHogenboom**
EASY-2100: add property:  bag-store.bag-name

#### When applied it will
* add property:  bag-store.bag-name
#### Related pull requests on github
repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-specs                      | [PR#81](https://github.com/DANS-KNAW/easy-specs/pull/81)     | ..

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.deposit/DepositDir.scala [hunk 1]
```diff
@@ -229,6 +229,7 @@ object DepositDir {
       addProperty("curation.performed", "no")
       addProperty("identifier.dans-doi.registered", "no")
       addProperty("identifier.dans-doi.action", "create")
+      addProperty("bag-store.bag-name", "bag")
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.deposit/DepositDir.scala]: is there some kind of constant or function with this name. Given that it is a hardcoded name here, indicates it should be somewhere where the bag is created as well.

@jo-pol sidenote: I would rather have a better name for the bag directory. Do you have any suggestions?

Comment (jo-pol) [reply src/main/scala/nl.knaw.dans.easy.deposit/DepositDir.scala]: Introduce a constant also to be used at https://github.com/DANS-KNAW/easy-deposit-api/blob/e0ddf23b3ac32591233306e15de0e112e5e4b0c5/src/main/scala/nl.knaw.dans.easy.deposit/DepositDir.scala#L45

The property name is indeed confusing for me in this context but is related to https://github.com/DANS-KNAW/easy-manage-deposit/pull/51

Comment (jo-pol) [reply src/main/scala/nl.knaw.dans.easy.deposit/DepositDir.scala]: To make it more complicated: 

Though the values are the same, you should share a constant with https://github.com/DANS-KNAW/easy-deposit-api/blob/c51c58a3b6bd24d87e295bf6dfef1f4ab84c2466/src/main/scala/nl.knaw.dans.easy.deposit/Submitter.scala#L92 in theory the value on line 45 could be different...

File: src/main/scala/nl.knaw.dans.easy.deposit/DepositDir.scala [hunk 2]
```diff
@@ -42,7 +42,7 @@ import scala.util.{ Failure, Success, Try }
  * @param id        the ID of the deposit
  */
 case class DepositDir private(draftBase: File, user: String, id: UUID) extends DebugEnhancedLogging {
-  val bagDir: File = draftBase / user / id.toString / "bag"
+  val bagDir: File = draftBase / user / id.toString / badDirName
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.deposit/DepositDir.scala]:
```suggestion
  val bagDir: File = draftBase / user / id.toString / bagDirName
```
Also in all other usages and declarations, lol

File: src/test/scala/nl.knaw.dans.easy.deposit/DepositDirSpec.scala [hunk 3]
```diff
@@ -53,13 +53,13 @@ class DepositDirSpec extends TestSupportFixture with MockFactory {
...
         (dir / "deposit.properties") should exist
-        (dir / "bag") should exist
-        (dir / "bag/bag-info.txt") should exist
-        (dir / "bag/bagit.txt") should exist
-        (dir / "bag/manifest-sha1.txt") should exist
-        (dir / "bag/tagmanifest-sha1.txt") should exist
-        (dir / "bag/data") should exist
-        (dir / "bag/metadata") should exist
+        (dir / bagDirName) should exist
+        (dir / s"$bagDirName/bag-info.txt") should exist
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/DepositDirSpec.scala]: while you're at it, make these like:

```suggestion
        (dir / bagDirName / "bag-info.txt") should exist
```

File: src/test/scala/nl.knaw.dans.easy.deposit/servlets/DoiSpec.scala [hunk 4]
```diff
@@ -21,7 +21,7 @@ import org.eclipse.jetty.http.HttpStatus._
 class DoiSpec extends DepositServletFixture {
   private def propsFile(uuid: String): File = testDir / "drafts/foo" / uuid / "deposit.properties"
 
-  private def jsonFile(uuid: String): File = testDir / "drafts/foo" / uuid / "bag/metadata/dataset.json"
+  private def jsonFile(uuid: String): File = testDir / "drafts/foo" / uuid / s"$bagDirName/metadata/dataset.json"
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/servlets/DoiSpec.scala]: same here

File: src/test/scala/nl.knaw.dans.easy.deposit/servlets/NotFoundSpec.scala [hunk 5]
```diff
@@ -108,7 +108,7 @@ class NotFoundSpec extends DepositServletFixture {
 
   private def propsFile(uuid: String): File = testDir / "drafts/foo" / uuid / "deposit.properties"
 
-  private def jsonFile(uuid: String): File = testDir / "drafts/foo" / uuid / "bag/metadata/dataset.json"
+  private def jsonFile(uuid: String): File = testDir / "drafts/foo" / uuid / s"$bagDirName/metadata/dataset.json"
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/servlets/NotFoundSpec.scala]: and here

File: src/test/scala/nl.knaw.dans.easy.deposit/servlets/UploadSpec.scala [hunk 6]
```diff
@@ -93,7 +93,7 @@ class UploadSpec extends DepositServletFixture {
     ))
     val uuid = createDeposit
     val relativeTarget = "path/to/dir"
-    val absoluteTarget = testDir / "drafts" / "foo" / uuid.toString / "bag/data" / relativeTarget
+    val absoluteTarget = testDir / "drafts" / "foo" / uuid.toString / s"$bagDirName/data" / relativeTarget
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/servlets/UploadSpec.scala]: and everywhere in this file...

**Pull request merged**